### ISSUE TITLE
Special characters with webservice target and WebApi

### ISSUE THREAD

**Issue by chwebdude** [NLog/NLog#1724]
**Question**
I would like to send the logdata to a Asp.Net WebApi Backend. But all special characters are encoded wrong. I tried to change the formatters in WebApi and tried all possible combinations of the `HttpPost` protocol (encoding and BOM) of the `WebService` target. But still the special characters are received as the unknown unicode character �.
What can I do?

**NLog version**: 4.3.9

**Platform**: .Net 4.6

**Current NLog config**

```
<target xsi:type='WebService' 
   name='ws' 
   url='http://localhost:8084/Logs' 
   protocol='HttpPost' 
   encoding='utf-8' 
   includeBOM='false'>
...
</target>
```

**WebApi Controller:**

```
[HttpPost]
public async Task<HttpResponseMessage> Post([FromBody] LogEntry logEntry)...
```

**Example**
**_Test Application**_
`logger.Debug("a b c é k è ï ?");`
**_Result**_
![image](https://cloud.githubusercontent.com/assets/7746357/19769716/df34c766-9c5c-11e6-9130-668956b202c6.png)

**Comment by 304NotModified** [comment]
Checked the code in NLog (in short) and it should work. Any change to check it with fiddler?

**Comment by 304NotModified** [comment]
I could reproduce the issue. See https://github.com/NLog/NLog/pull/1725

I looks like the problem is in `UrlHelper.UrlEncode`, see [source](https://github.com/NLog/NLog/blob/0b79e8b1019b6b4a4a3f1f059a8cb642b34d5760/src/NLog/Internal/UrlHelper.cs#L69-L85)

not sure how to fix.

**Comment by 304NotModified** [comment]
I think I could use `Uri.EscapeDataString` ?

**Comment by chwebdude** [comment]
Resolved with #1735

**Comment by 304NotModified** [comment]
Thanks! We will close this one as it has been merged :)
